lib/stats_reporter.rb

```diff
@@ -0,0 +1,117 @@
+
+  def self.df_output
+    stdout_str, _status = Open3.capture2('df -h')
+    stdout_str
+  end
+
+  def self.storage_mount_lines
+    df_output.split("\n").select { |line| line.include?('services-disk') }
+  end
+
+  def self.parsed_storage_mount_lines
+    storage_mount_lines.map do |line|
+      storage_root = {}
+      split_line = line.split(' ')
+      storage_root[:total] = split_line[0]
+      storage_root[:used] = split_line[1]
+      storage_root[:remaining] = split_line[2]
+      storage_root[:percent_used] = split_line[3]
+      storage_root[:filesystem] = split_line[4]
+      storage_root[:percent_free] = "#{(100 - split_line[3].chop.to_i)}%"
+      storage_root
+    end
+  end
+
+  def self.storage_report_lines
+    parsed_storage_mount_lines.map do |h|
+      [h[:filesystem], h[:total], h[:used], h[:percent_used], h[:remaining], h[:percent_free]]
+    end
+  end
+
+  private_class_method :storage_mount_lines, :parsed_storage_mount_lines, :storage_report_lines
+
+  private
+
+  def ssl_options
+    {
+      client_cert: OpenSSL::X509::Certificate.new(File.read(Settings.ssl.cert_file)),
+      client_key:  OpenSSL::PKey::RSA.new(File.read(Settings.ssl.key_file), Settings.ssl.key_pass)
+    }
+  end
+
+  def conn
+    @conn ||= if ENV['ROBOT_ENVIRONMENT'] == 'test'
+                Faraday.new(url: Settings.workflow.url)
+              else
+                Faraday.new(url: Settings.workflow.url, ssl: ssl_options) do |f|
+                  f.adapter :net_http_persistent
+                end
+              end
+  end
+
+  def waiting_count
+    # in sdr-pc, this relied on register-sdr, which isn't a step in the WF
+    # so I'm guessing at the appropriate waiting parameter value
+    response = conn.get "workflow_queue?repository=#{REPOSITORY}&workflow=#{INGEST_WF}&waiting=start-ingest&count-only=true"
+    Nokogiri::XML(response.body).at_xpath('/objects/@count').value.to_i
+  end
+
+  def erroring_count
+    errors = 0
+    INGEST_WF_STEPS.each do |step|
+      response = conn.get "workflow_queue?repository=#{REPOSITORY}&workflow=#{INGEST_WF}&error=#{step}&count-only=true"
+      errors += Nokogiri::XML(response.body).at_xpath('/objects/@count').value.to_i
+    end
+    errors
+  end
+
+  def completed_count
+    # in sdr-pc, this relied on ingest-cleanup, which isn't a step in the WF
+    # so I'm guessing at the appropriate completed parameter value
+    response = conn.get "workflow_queue?repository=#{REPOSITORY}&workflow=#{INGEST_WF}&completed=complete-ingest&count-only=true"
+    Nokogiri::XML(response.body).at_xpath('/objects/@count').value.to_i
+  end
+
+  def archived_count
+    response = conn.get "workflow_archive?repository=#{REPOSITORY}&workflow=#{INGEST_WF}&count-only=true"
+    Nokogiri::XML(response.body).at_xpath('/objects/@count').value.to_i
+  end
+
+  def workflow_report_lines
+    [[
+      INGEST_WF,
+      waiting_count,
+      erroring_count,
+      completed_count,
+      archived_count
+    ]]
+  end
+end
```

Comment:
I think it is AWESOME that this rewrite of the code made a single 117 line file containing a single class.